progression/src/main/java/me/mykindos/betterpvp/progression/profession/skill/woodcutting/TreeCompactor.java

```diff
@@ -0,0 +1,99 @@
+package me.mykindos.betterpvp.progression.profession.skill.woodcutting;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import lombok.Getter;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.progression.Progression;
+import me.mykindos.betterpvp.progression.profession.skill.ProgressionSkillDependency;
+import me.mykindos.betterpvp.progression.profile.ProfessionProfile;
+import me.mykindos.betterpvp.progression.profile.ProfessionProfileManager;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+
+@Singleton
+@BPvPListener
+public class TreeCompactor extends WoodcuttingProgressionSkill implements Listener {
+    private final ProfessionProfileManager professionProfileManager;
+
+    @Getter
+    private double cooldown;
+
+    @Inject
+    public TreeCompactor(Progression progression, ProfessionProfileManager professionProfileManager) {
+        super(progression);
+        this.professionProfileManager = professionProfileManager;
+    }
+
+    @Override
+    public String getName() {
+        return "Tree Compactor";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[] {
+                "You gain access to the <green>/treecompactor</green> command",
+                "",
+                "This command lets you turn a stack of logs",
+                "into a singular log"
+        };
+    }
+
+    @Override
+    public Material getIcon() {
+        return Material.OAK_WOOD;
+    }
+
+    @Override
+    public boolean isGlowing() {
+        return true;
+    }
+
+    @Override
+    public void loadConfig() {
+        super.loadConfig();
+        cooldown = getConfig("cooldown", 20.0, Double.class);
+    }
+
+    @Override
+    public ProgressionSkillDependency getDependencies() {
+        final String[] dependencies = new String[]{"Bark Bounty"};
+        return new ProgressionSkillDependency(dependencies, 100);
+    }
+
+    // no dependencies yet
+
+    /**
+     * @return the player's skill level
+     */
+    public int getPlayerSkillLevel(Player player) {
+        Optional<ProfessionProfile> profile = professionProfileManager.getObject(player.getUniqueId().toString());
+
+        return profile.map(this::getPlayerSkillLevel).orElse(0);
+    }
+
+    /**
+     * This function's purpose is to return a boolean that tells you if the player has the skill
+     * <b>No More Leaves</b>
+     */
+    public boolean doesPlayerHaveSkill(Player player) {
+        return getPlayerSkillLevel(player) > 0;
+    }
+
+    @EventHandler
+    public void onPlaceCompactedLog(BlockPlaceEvent event) {
+        if (!event.getBlock().getType().equals(Material.OAK_WOOD)) return;
```

Comment:
Youl want to add a check here for client administration mode, we wouldnt want to stop admins from placing them

client#isAdministrating()